design-proposals/host-ip-pool.md

```diff
@@ -0,0 +1,442 @@
+  - It's though for Metal3 baremetal.
+- Azure private addresses: https://docs.microsoft.com/en-us/azure/virtual-network/ip-services/private-ip-addresses
+  - Specific to azure
+  - Some code has to be put in place to consume it.
+
+### User Stories
+
+- As a cluster administrator, I want to configure static IPs at non control-plane NICs on a pool of nodes without creating one per each node.
+- As a cluster administrator, I want scale up the cluster with new nodes getting IPs from the policy.
+- As a cluster administrator, I want each of these static IPs to be unique, and taken from a defined range, and freed when unused.
+
+### Goals
+
+- Allow users to use NodeNetworkConfigurationPolicy to configure static cluster-unique IP addresses without hard-coding them.
+
+### Non-Goals
+
+- This design does not cover IP address configuration of management NIC on day-0/1
+
+## Proposal
+
+### User roles
+
+**cluster admin** is a user responsible for managing the cluster node
+networking.
+
+### Workflow Description (assign static address with IPPool)
+
+1. The cluster admin creates `NodeIPPool` resources with IP ranges and a name.
+2. The cluster admin configures node network interfaces statically, with IPs from the `NodeIPPool` using a
+`NodeNetworkConfigurationPolicy` referencing them.
+3. The cluster admin waits for the `NodeNetworkConfigurationPolicy` to succeed
+
+#### Variation (exhausted range at the NodeIPPool)
+
+When the IP pool does not have an available IP address, the `NNCP` will fail
+with a explicit message, the `NodeIPPool` has to be re-created with a bigger
+range.
+
+### Workflow Description (removing a reference to NodeIPPool by NNCP re-creation)
+
+1. The cluster admin deletes and creates the NNCP (NNCP with captures
+   are not allowed to be modified).
+2. The cluster admin waits for the NodeNetworkConfigurationPolicy to succeed.
+   the interfaces will have re-allocated IPs.
+
+### Workflow Description (scaling up cluster)
+
+1. The cluster admin adds a new node affected by the `NodeNetworkConfigurationPolicy`.
+2. The cluster admin waits for the `NodeNetworkConfigurationPolicy` to succeed, causing the new node to take the address from the `NodeIPPool`
+
+### Workflow Description (scaling down cluster)
+
+1. The cluster admin removes a node that was taking an address from a `NodeIPPool`.
+2. The cluster admin waits for the `NodeNetworkConfigurationPolicy` to succeed, thus freeing the IPs from the removed node.
+
+### Workflow Description (manual NodeIPPool clean up)
+
+1. The cluster admin finds some stale IPs, They are inconsistencies between `NodeIPPool.status` and `NodeNetworkConfigurationEnactment.status`.
+2. The cluster admin resets the `NodeIPPool` following instructions at "API Exntensions/Manual Garbage Collection" chapter.
+3. The cluster admin waits for `NodeIPPool` status to succeed
+
+NOTE: This is a manual garbage collector that is good enough for first implementation it can be converted into automatic in the future.
```

Comment:
```suggestion
**NOTE**: This is a manual garbage collector that is good enough for first implementation it can be converted into automatic in the future.
```